pandas/core/generic.py

```diff
@@ -11403,6 +11388,18 @@ def last_valid_index(self):
         return self._find_valid_index("last")
 
 
+def _set_join_index(
+    left: FrameOrSeries, right: FrameOrSeries, join_index: Index
+) -> Tuple[FrameOrSeries, FrameOrSeries]:
```

Comment:
do we know that type(left) == type(right)?  if so, should this be FrameOrSeriesT?

Reply:
What's `FrameOrSeriesT`? I just tried searching for it in the codebase but couldn't find it